### PR TITLE
Fix vue settings checks and filenames

### LINKED ISSUE #11794: Vue check script does not work with split output
See https://github.com/nextcloud/server/pull/11793

0.js 1.js etc are not checked for differences.
Also. I must admit I don't like those names :P

@skjnldsv please have a look
CC: @juliushaertl @ChristophWurst as our other vue overlords

### PR DESCRIPTION
**Pull request by skjnldsv**
- Fix the vue settings tests
- Better file naming of the compiled files
- Fixes #11794

### CODE REVIEW

**Comment by skjnldsv** [comment]
> Comparing ./settings/js to ./settings/js.orig
> ./settings/js build is up-to-date

Damn, it works here :/

**Comment by MorrisJobke** [comment]
@skjnldsv What is this about?

**Comment by skjnldsv** [comment]
> @skjnldsv What is this about?

I updated OP

**Comment by MorrisJobke** [comment]
CI vue test is failing :/

**Comment by skjnldsv** [comment]
> CI vue test is failing :/

Yes, It passes on my setup, but not on drone.
My script must be failing I guess

**Comment by rullzer** [comment]
@skjnldsv kill or?

**Comment by skjnldsv** [comment]
@rullzer Well, we still should have this, but my bash-fu is failing me :)

**Comment by juliushaertl** [comment]
@skjnldsv Simply using something like https://github.com/nextcloud/files_videoplayer/blob/18649fea117e44f2ed3bbba5075a89361c2acd1b/.drone.yml#L10-L15 should already do the trick :wink:

**Comment by skjnldsv** [comment]
@juliushaertl so much better!!!! :hugs:

**Comment by skjnldsv** [comment]
Rebased and ready to review :)

File: .drone.yml [hunk 1]
```diff
@@ -15,9 +15,13 @@ pipeline:
       matrix:
         TESTS: jsunit
   vue-build-settings:
-    image: node
+    image: nextcloudci/node:node-1
```
Comment (MorrisJobke) [inline .drone.yml]:
```suggestion
    image: nextcloudci/node:node-4
```

This is the name 🙈 https://hub.docker.com/r/nextcloudci/node/tags

**Comment by MorrisJobke** [comment]
With all those file name changes also add all the files to `settings/.l10nignore` (regex is not supported there yet).

**Comment by skjnldsv** [comment]
> With all those file name changes also add all the files to `settings/.l10nignore` (regex is not supported there yet).

What do we need that? :thinking:

**Comment by MorrisJobke** [comment]
> What do we need that? 🤔

To ignore parsing of compiled files, because it's very likely that they contain a `t()` which is then extracted as translation even if it is just a minified function name ;)

**Comment by rullzer** [comment]
Maybe it is easier to output the compiled files into a special folder we can then ignore right?

**Review by nickvergessen — APPROVED** [review]

**Comment by MorrisJobke** [comment]
> Maybe it is easier to output the compiled files into a special folder we can then ignore right?

Not yet implemented but would also work.

**Comment by skjnldsv** [comment]
@juliushaertl it was not the map config so it seems :(

**Comment by skjnldsv** [comment]
> To ignore parsing of compiled files, because it's very likely that they contain a `t()` which is then extracted as translation even if it is just a minified function name ;)

Which is fine because we need those translations, right? ^^
The vue fix we did lately is not 100% proof. I'd rather have both for now :)

**Comment by skjnldsv** [comment]
Failures unrelated! Good to go :tada: :ship:

**Review by rullzer — APPROVED** [review]

**Pull request merged**